custom_components/nano_pk/sensor.py

```diff
@@ -30,6 +31,8 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         for p in bridge.data().values(): 
             if p.key()=="Störung": 
                 entities.append(HargassnerErrorSensor(bridge, name))
+                entities.append(HargassnerProblemSensor(bridge, name))
```

Comment:
This will not work, I think: if the "Störung" flag is set, the corresponding error id  is passed as "Störungs Nr". However, there are no corresponding messages for problems or solutions (at least I'm not aware of it), so these categories cannot be handled properly. The new sensors you add process the same "Störungs Nr" as the ErrorSensor, so they will give you incorrect problems and solutions.